api/queries_issue.go

```diff
@@ -443,6 +450,44 @@ func IssueReopen(client *Client, repo ghrepo.Interface, issue Issue) error {
 	return err
 }
 
+func IssueTemplates(client *Client, repo *Repository) ([]IssueTemplate, error) {
+	type response struct {
+		Repository struct {
+			IssueTemplates   []IssueTemplate
+			HasIssuesEnabled bool
+		}
+	}
+
+	query := `
+	query IssueTemplates() {
+		repository(owner: $owner, name: $repo) {
+			issueTemplates
+			hasIssuesEnabled
+		}
+	}
+	`
+
+	variables := map[string]interface{}{
+		"owner": repo.RepoOwner(),
+		"repo":  repo.RepoName(),
+	}
+
+	// make the call, get response
+	var resp response
+	err := client.GraphQL(repo.RepoHost(), query, variables, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if !resp.Repository.HasIssuesEnabled {
```

Comment:
I would say that this check is outside the scope of responsibilities for this function. If this `api` call is being made then the caller should be responsible for making sure the repository has issue enabled.